plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/BaseSqlServerTypeMapping.java

```diff
@@ -331,6 +332,47 @@ public void testSqlServerNchar()
                 .execute(getQueryRunner(), sqlServerCreateAndInsert("test_sqlserver_char"));
     }
 
+    @Test
+    public void testSqlServerUniqueIdentifier()
+    {
+        SqlDataTypeTest.create()
+                .addRoundTrip("uniqueidentifier", "NULL", UUID, "CAST(NULL AS uuid)")
+                .addRoundTrip("uniqueidentifier", "'d5416a8a-5faf-208d-a0cc-aaf97732f49a'", UUID, "uuid 'd5416a8a-5faf-208d-a0cc-aaf97732f49a'")
+                .addRoundTrip("uniqueidentifier", "'00000000-0000-0000-0000-000000000000'", UUID, "uuid '00000000-0000-0000-0000-000000000000'")
+                .addRoundTrip("uniqueidentifier", "'ffffffff-ffff-ffff-ffff-ffffffffffff'", UUID, "uuid 'ffffffff-ffff-ffff-ffff-ffffffffffff'")
+                .addRoundTrip("uniqueidentifier", "'0E984725-C51C-4BF4-9960-E1C80E27ABA0wrong'", UUID, "uuid '0E984725-C51C-4BF4-9960-E1C80E27ABA0'")
```

Comment:
Does this case fail with `trinoCreateAndInsert`/`trinoCreateAsSelect`?

Reply:
I suppose such characters aren't allowed at the engine level.

Reply:
indeed. Marking as resolved.